File: android/src/main/kotlin/com/benjamin/horner/flutter_acs_card_reader/notifiers/DataTransferNotifier.kt
package com.benjamin.horner.flutter_acs_card_reader

import io.flutter.plugin.common.MethodChannel
import org.json.JSONObject

class DataTransferNotifier {
    fun updateState(data: Map<String, Any>, channel: MethodChannel) {
        val json = JSONObject(data as Map<*, *>).toString()
        channel.invokeMethod("onReceiveDataEvent", json)
    }
}

File: ios/Classes/Notifiers/DataTransferNotifier.swift
//
//  DataTransferNotifier.swift
//  flutter_acs_card_reader
//
//  Created by Benjamin Horner on 24/11/2023.
//

import Flutter

class DataTransferNotifier {
    func updateState(data: [String: Any], channel: FlutterMethodChannel) {
        if let jsonData = try? JSONSerialization.data(withJSONObject: data),
           let jsonString = String(data: jsonData, encoding: .utf8) {
            channel.invokeMethod("onReceiveDataEvent", arguments: jsonString)
        }
    }
}
